src/dotty/tools/dotc/transform/CheckIllegalOverrides.scala

```diff
@@ -0,0 +1,41 @@
+package dotty.tools.dotc
+package transform
+
+import TreeTransforms._
+import core._
+import Contexts._, Definitions._, Decorators._
+
+/** Checks if the user is trying to illegally override synthetic classes,
+  * primitives or their boxed equivalent
+  */
+class CheckIllegalOverrides extends MiniPhaseTransform {
```

Comment:
I wouldn't call this an override, it's more of a re-definition. But I'm also fine with the current name.